### PR TITLE
REALMC-7629: Add app and user integration tests for the Realm client

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: internal/cloud/realm/app.go [hunk 1]
```diff
@@ -8,14 +8,52 @@ import (
...
 
 	"github.com/10gen/realm-cli/internal/utils/api"
-
-	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
-var (
+const (
 	appsPathPattern = adminAPI + "/groups/%s/apps"
+	appPathPattern  = adminAPI + "/groups/%s/apps/%s"
```
Comment (UnicodeSnowman) [inline internal/cloud/realm/app.go:15]: [nit] could just use `appsPathPattern` in this

File: internal/cloud/realm/user.go [hunk 2]
```diff
@@ -4,26 +4,34 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/10gen/realm-cli/internal/utils/api"
+	"go.mongodb.org/mongo-driver/bson/primitive"
```
Comment (UnicodeSnowman) [inline internal/cloud/realm/user.go:10]: ☝️ need a linebreak between these two deps

File: internal/cloud/realm/user.go [hunk 3]
```diff
@@ -4,26 +4,34 @@ import (
...
 
 const (
-	apiKeysPathPattern = adminAPI + "/groups/%s/apps/%s/api_keys"
-	usersPathPattern   = adminAPI + "/groups/%s/apps/%s/users"
+	apiKeysPathPattern      = adminAPI + "/groups/%s/apps/%s/api_keys"
+	pendingUsersPathPattern = adminAPI + "/groups/%s/apps/%s/user_registrations/pending_users"
+	usersPathPattern        = adminAPI + "/groups/%s/apps/%s/users"
+	userPathPattern         = adminAPI + "/groups/%s/apps/%s/users/%s"
+	userDisablePathPattern  = adminAPI + "/groups/%s/apps/%s/users/%s/disable"
+	userLogoutPathPattern   = adminAPI + "/groups/%s/apps/%s/users/%s/logout"
```
Comment (UnicodeSnowman) [inline internal/cloud/realm/user.go:19]: [nit] same here, could avoid some copy-pasta

**Review by UnicodeSnowman — APPROVED** [review]
just nits, LGTM

**Review by kpatel71716 — APPROVED** [review]
LGTM! Great stuff!

**Pull request merged**